rust-mode.el

```diff
@@ -573,6 +584,46 @@ the desired identifiers), but does not match type annotations \"foo::<\"."
             ((not (looking-at (rx (0+ space) "<")))
 	     (throw 'rust-path-font-lock-matcher match))))))))
 
+(defun rust-next-string-interpolation (limit)
+  "Search forward from point for next Rust interpolation marker
+before LIMIT.
+Set point to the end of the occurrence found, and return match beginning
+and end."
+  (while (search-forward "{{" limit t)
```

Comment:
It seems to me that this approach can cause the function to skip over formats that come before escapes.  The test would be something like `"{} {{"`.  I think instead there has to be just one loop that looks at the following character to decide what to do.